misk/src/test/kotlin/misk/web/GracefulShutdownTest.kt

```diff
@@ -0,0 +1,124 @@
+package misk.web
+
+import com.google.common.util.concurrent.AbstractIdleService
+import com.google.common.util.concurrent.ServiceManager
+import com.google.inject.util.Modules
+import misk.MiskTestingServiceModule
+import misk.ReadyService
+import misk.ServiceModule
+import misk.inject.KAbstractModule
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import misk.time.ClockModule
+import misk.web.actions.WebAction
+import misk.web.jetty.JettyService
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import wisp.logging.getLogger
+import java.io.IOException
+import java.time.Duration
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@MiskTest(startService = true)
+internal class GracefulShutdownTest {
```

Comment:
Could you add documentation for what this test is attempting to validate and how it's doing that?